### PR TITLE
Resolve #33 -- Summoner Spells

### LINKED ISSUE #33: Add summoner spells
Add Ignite, Ghost, Revive, Smite, Clarity and Exhaust

### PR DESCRIPTION
**Pull request by Dawiss2**
Added all summoner spells that are possible atm (Ignite, Ghost, Revive, Smite, Clarity, Exhaust)

Refs #33

### CODE REVIEW

**Review by Deudly — CHANGES_REQUESTED** [review]
Please, follow the committing guidelines https://github.com/LeagueSandbox/GameServer/wiki/4.7.-Committing
By the way, if someone damages you, the Recall casting should stop

File: Champions/Global/SummonerHaste.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        public void OnStartCasting(Champion owner, Spell spell, Unit target)
+        {
+            ChampionStatModifier statMod = new ChampionStatModifier();
+            statMod.MoveSpeed.PercentBonus = 27 / 100.0f;
+            owner.AddStatModifier(statMod);
+            var HasteBuff = ApiFunctionManager.AddBuffHUDVisual("SummonerHaste", 10.0f, 1, owner, 10.0f);
+            ApiFunctionManager.CreateTimer(10.0f, () =>
+            {
+                owner.RemoveStatModifier(statMod);
+
```
Comment (MythicManiac) [inline Champions/Global/SummonerHaste.cs]: Unnecessary blank line

File: Champions/Global/SummonerHeal.cs [hunk 2]
```diff
@@ -75,10 +81,17 @@ public void OnFinishCasting(Champion owner, Spell spell, Unit target)
...
-            //addBuff(buff2)
-            
+            ApiFunctionManager.AddBuffHUDVisual("SummonerHeal", 1.0f, 1, owner, 1.0f);
+            ChampionStatModifier statMod = new ChampionStatModifier();
+            statMod.MoveSpeed.PercentBonus = 30 / 100.0f;
+            owner.AddStatModifier(statMod);
+            ApiFunctionManager.CreateTimer(1.0f, () =>
+            {
+                owner.RemoveStatModifier(statMod);
+
```
Comment (MythicManiac) [inline Champions/Global/SummonerHeal.cs]: Unnecessary blank lines

File: Champions/Global/SummonerMana.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                if (owner.Team == value.Team && i != 0)
+                {
+                    var currentMana = value.GetStats().CurrentMana;
+                    maxMana = value.GetStats().ManaPoints.Total;
+
+                    if (currentMana * 100 / maxMana < lowestManaPercentage && owner != value)
+                    {
+                        lowestManaPercentage = currentMana * 100 / maxMana;
+                        nearbychampion = value;
+
```
Comment (MythicManiac) [inline Champions/Global/SummonerMana.cs]: Unnecessary blank line

File: Champions/Global/SummonerMana.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                ApiFunctionManager.AddParticleTarget(nearbychampion, "global_ss_clarity_02.troy", nearbychampion);
+            }
+
+            var mp = owner.GetStats().CurrentMana;
+            var maxMp = owner.GetStats().ManaPoints.Total;
+            if (mp + maxMp * PERCENT_MAX_MANA_HEAL < maxMp)
+                owner.GetStats().CurrentMana = mp + maxMp * PERCENT_MAX_MANA_HEAL;
+            else
+                owner.GetStats().CurrentMana = maxMp;
+
```
Comment (MythicManiac) [inline Champions/Global/SummonerMana.cs]: Unnecessary blank line (you have 2 in a row, remove at least the other :) )

**Review by MythicManiac — CHANGES_REQUESTED** [review]
Nice work 👍 , though there are some issues

Aside from my comments, this doesn't follow the commit guidelines as @Deudly pointed out. (Your pull request has the proper format, but the commits you want to merge do not).

The format can be fixed by squash merging, but that leaves the recall issue.

I'd say we have 4 options here, which are:
1. Create an issue for the recall, squash merge this as it is now
2. Fix the recall issue in this PR, squash merge
3. Fix the recall issue and your commit format, merge
4. Fix your commit format and create an issue for recall, merge

Up to you which one, though I'd prefer you address my comments regardless which option you choose :)

**Review by MythicManiac — APPROVED** [review]

**Comment by MythicManiac** [comment]
Issue created to #32, merging

**Pull request merged**